dialogflow-cx/streaming_detect_intent_infinite.py

```diff
@@ -0,0 +1,720 @@
+            logger.debug("End of response stream in listen_print_loop")
+            return True  # Restart loop if the stream ends without end_interaction
+        except asyncio.TimeoutError:
+            logger.warning("Timeout waiting for response in listen_print_loop")
+            continue
+        except Exception as e:
+            logger.error(f"Error in listen_print_loop: {e}")
+            break
+
+    return True  # Default to restarting the loop
+
+
+async def handle_audio_input_output(
+    dialogflow_streaming: DialogflowCXStreaming,
+    audioIO: AudioIO,
+    audio_queue: asyncio.Queue,
+) -> None:
+    """Handles audio input and output concurrently."""
+
+    async def cancel_push_task(push_task: Optional[asyncio.Task]) -> None:
+        """Helper function to cancel push task safely."""
+        if push_task is not None and not push_task.done():
+            push_task.cancel()
+            try:
+                await push_task
+            except asyncio.CancelledError:
+                logger.debug("Push task cancelled successfully")
+
+    push_task: Optional[asyncio.Task] = None
+    try:
+        async with asyncio.TaskGroup() as tg:
+            push_task = tg.create_task(
+                push_to_audio_queue(audioIO.generator(), audio_queue)
+            )
+            while True:
+                responses: AsyncGenerator[
+                    dialogflowcx_v3.StreamingDetectIntentResponse, None
+                ] = dialogflow_streaming.streaming_detect_intent(audio_queue)
+                should_continue: bool = await listen_print_loop(
+                    responses,
+                    audioIO,
+                    audio_queue,
+                    dialogflow_streaming.dialogflow_timeout,
+                )
+                if not should_continue:
+                    logger.debug(
+                        "End interaction detected, exiting handle_audio_input_output"
+                    )
+                    await cancel_push_task(push_task)
+                    break
+                logger.debug("Restarting listen_print_loop")
+    except asyncio.CancelledError:
+        logger.warning("Handling of audio input/output was cancelled.")
+        await cancel_push_task(push_task)
+    except Exception as e:
+        logger.error(f"An unexpected error occurred: {e}")
+
+
+async def main(
+    agent_name: str,
+    debug: bool,
+    model: str,
+    language_code: str,
+    dialogflow_timeout: float,
+    sample_rate: int,
+    voice: Optional[str],
+    single_utterance: bool,
+) -> None:
+    """Start bidirectional streaming from microphone input to speech API"""
+
+    chunk_size: int = int(sample_rate * CHUNK_SECONDS)
+
+    audioIO: AudioIO = AudioIO(sample_rate, chunk_size)
+    dialogflow_streaming: DialogflowCXStreaming = DialogflowCXStreaming(
+        agent_name,
+        language_code,
+        model,
+        dialogflow_timeout,
+        debug,
+        sample_rate,
+        "LINEAR16",
```

Comment:
Since these LINEAR16 are hardcoded and not configurable, can we remove them from `DialogCXStreaming` altogether and inline them directly where they're used?